youtube_dl/extractor/brightcove.py

```diff
@@ -508,7 +510,21 @@ def _extract_urls(webpage):
                 'http://players.brightcove.net/%s/%s_%s/index.html?videoId=%s'
                 % (account_id, player_id, embed, video_id))
 
-        return entries
+        # 	<video data-brightcove-video-id="5320421710001" data-account="245991542" data-player="SJWAiyYWg" data-embed="default" class="video-js" controls itemscope itemtype="http://schema.org/VideoObject">
+        for video in re.findall(r'(?i)(<video[^>]+>)', webpage):
+            attrs = extract_attributes(video)
+                
+            video_id    = dict_get(attrs, ['data-brightcove-video-id', 'data-video-id'])
+            account_id  = attrs.get('data-account')
+            player_id   = attrs.get('data-player')
+            embed       = attrs.get('data-embed')
+
+            if video_id and account_id and player_id and embed:
+                entries.append(
+                    'http://players.brightcove.net/%s/%s_%s/index.html?videoId=%s'
+                    % (account_id, player_id, embed, video_id))
+
+	return entries
```

Comment:
Done. I have to say, it seems to have both more code duplication (from brightcove.py and generic.py both) as well as more fragile (what happens when there are other kinds of videos on bostonglobe.com that the Generic extractor would have caught, but this one won't)...just shows, once again, I don't understand the youtube-dl philosophy...maybe that could be better documented.